ran-core/src/test/java/io/ran/TestDoublePerfTest.java

```diff
@@ -0,0 +1,118 @@
+package io.ran;
+
+import io.ran.testclasses.Brand;
+import io.ran.testclasses.IndexedCar;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class TestDoublePerfTest {
+
+	public static final int NUMBER_OF_SUVS = 50;
+	public static final int NUMBER_OF_CARS = 2000;
+	private TestCarRepo carRepo;
+	List<String> carIds = new ArrayList<>();
+
+	@Before
+	public void setup() {
+		GuiceHelper helper = new GuiceHelper();
+
+		carRepo = helper.injector.getInstance(TestCarRepo.class);
+
+
+		for (int i = 0; i < NUMBER_OF_CARS; i++) {
+			IndexedCar car = new IndexedCar();
+			car.setBrand(Brand.Porsche);
+			car.setConstructionDate(ZonedDateTime.now().minus(Duration.ofDays(2)));
+			car.setCrashRating(2.0);
+			car.setId(UUID.randomUUID().toString());
+			car.setTitle("SUV " + ((int) (Math.random() * NUMBER_OF_SUVS)));
+
+			carRepo.save(car);
+			carIds.add(car.getId());
+		}
+
+	}
+
+	@Test
+	public void perf() {
+		long s = System.currentTimeMillis();
+		for (int i = 0; i < 1000; i++) {
+			Optional<IndexedCar> foundCar = carRepo.query().eq(IndexedCar::getId, carIds.get(((int) (Math.random() * NUMBER_OF_CARS)))).execute().findFirst();
+			assertTrue(foundCar.isPresent());
+		}
+		System.out.println(System.currentTimeMillis() - s + "ms for pk");
```

Comment:
Should we be setting some upper extreme bound for performance? AKA fail if above ??? ms

Reply:
we ignore it instead, as we do with performance tests in general